### DISCUSSION TITLE
CP-Editor doesn't show me COMPILATION ERROR

### DISCUSSION THREAD

**Discussion by md-shamim-ahmad** [cpeditor/cpeditor#1151] [Q&A]
### Is there an existing issue for this?

- [ ] I have searched the existing issues

### Current Behavior

Hi,
I found an error when I try to solve the problem given under below.
![](https://user-images.githubusercontent.com/56692841/220165180-f6a9c057-d743-4f64-a634-36ba0e963eaa.png)
![](https://user-images.githubusercontent.com/56692841/220165212-78ab2971-6046-4b4b-89f7-b026c69a9c3d.png)
Basically the error is that,
Look at line number **```49```**, **```61```**, **```70```** of this code.
I declare a function whose return type is ```uint``` but I does not declare ```uint``` in previous into my code.
As I does not declare ```uint``` in previous into my code so I can simply say that my code give a  ```COMPILATION_ERROR```.
Because C++17 has no uint (unsigned int) so for that my give solution code will give me a ```COMPILATION_ERROR``` but ```CPEditor``` does not show me any error.
I found a documentation [here](https://stackoverflow.com/questions/65911537/what-does-uint-means-in-c-while-the-declaration-of-a-variable-in-class) that uint is not a part of C++.

### Expected Behavior

**Although ```CP Editor``` doesn't show me any error but ```Codeforces``` platform show ```COMPILATION_ERROR```**
![](https://user-images.githubusercontent.com/56692841/220167176-6836bdfb-0f01-47e9-928e-d26fce449ac0.png)
So my Expectation is that ```CP Editor``` also show me ```COMPILATION_ERROR```.

### Steps To Reproduce

[]()

### CP Editor Version

6.9.2

### OS

ubuntu 22.04

### Are you willing to contribute to this?

- [ ] I'm ready to fix this (and will open a PR soon)
- [ ] I'd like to have a try (with the help of the maintainers)
- [ ] No, thanks

**Comment by ouuan** [comment]
It's caused by the compiler you use. CP Editor doesn't compile your code, it calls the compiler to compile your code. So it's not related to CP Editor.

It seems that `uint` works on Linux but not on Windows. But it's not related to CP Editor, so I know no more than you.

**Reply by md-shamim-ahmad** [reply]
But If I change return type ```ull``` to ```uint``` at line 65 then the output will be not correct.
It provides me a garbage value.
If it works on Ubuntu then why does it give me garbage values when I change return type ```ull``` to ```uint``` at line 65.
And I also do not know that it works good or not in windows.

**Reply by ouuan** [reply]
I know no more than you as it's not related to CP Editor. Please try searching online.

`uint` is `unsigned int`, while `ull` is `unsigned long long`. I don't know what you mean by "garbage value", but they are of different sizes.
